### ISSUE TITLE
[JsonParse] Unable to parse JSON com.fasterxml.jackson.core.JsonParseException: Unexpected character ('*' (code 42))

### ISSUE THREAD

**Issue by Drugoy** [Graylog2/graylog2-server#8685]
Getting errors about unexpected chars when using a pipeline that makes use of a rule with `parse_json(to_string($message.field))`.

## Expected Behavior

Valid JSON objects should not produce errors.
Having symbols like wildcard is absolutely legal inside values in JSON objects.

## Current Behavior

Getting errors about unexpected characters.

## Possible Solution

The bug looks like due to a wrong use of the 3rd-party lib.

## Steps to Reproduce (for bugs)

1. Have a pipeline with just 1 stage with just 1 rule attached to just 1 steam (we use GELF UDP).
The rule source should be:
```
rule "JSON extractor"
when
    has_field("message_body")
    AND
    starts_with(to_string($message.message_body), "{")
    AND
    ends_with(to_string($message.message_body), "}")
then
    set_fields(to_map(parse_json(to_string($message.message_body))), "msg_body_json_");
end
```
2. Send 1 message to said stream so that `message_body` field would contain a JSON.
Said JSON has to contain a wildcard in value for any key:
`{"key":"va*lue"}`

## Context

Due to another bug/missing feature in Graylog - there's no easy way to 'parse' a composite message (consisting of regular text + data in JSON) into separate fields, so that fields of message in Graylog would be taken from both simple string and from JSON of the input message.

## Your Environment

* Graylog Version: 3.3.3+88d0659
* Elasticsearch Version: 6.8.5-1
* MongoDB Version: 4.0.13-1.el7.x86_64
* Operating System: CentOS 7 Linux 3.10.0-1127.el7.x86_64
* Browser version: Chromium 80.0.3968.0 x64

**Comment by bernd** [comment]
@Drugoy I used the exact pipeline you posted and tried to reproduce it by sending the following message:

```
cat <<__GELF | ncat -u 127.0.0.1 12201
{"short_message":"hello2","_message_body":"{\"key\":\"va*lue\"}"}
__GELF
```

The message gets parsed correctly and shows up in the search:

![image](https://user-images.githubusercontent.com/461/89298421-9f3f7b00-d665-11ea-9ab6-f688b5b9808c.png)

So the JSON parsing works as expected. I guess there is an issue with quoting the JSON in the GELF message and graylog receives invalid JSON and thus cannot parse it.

**Comment by Drugoy** [comment]
Bug was incorrectly filed.
When I was filing a bug I reduced/simplified the example.
Turns out the error was in the part that got removed while reducing the example: the input JSON indeed was incorrect (without double quotes) and somehow I hadn't noticed it.
The input JSON was a result of JSON extraction: turns out extracted JSONs lose double quotes, even though the original JSON (before extraction) had them.
Unfortunately, such behavior blocks 'chaining' JSON extractors in Graylog.
